snapcraft/parts/plugins/_ros.py

```diff
@@ -127,6 +132,49 @@ def _get_build_commands(self) -> List[str]:
         specific functionality.
         """
 
+    def _get_list_packages_commands(self) -> List[str]:
+        """Generate a list of ROS 2 packages available in build snaps.
+
+        The ROS 2 workspaces contained in build snaps are crawled with `rospack`
+        to establish the list of all available ROS 2 packages.
+        The package names are then resolved with `rosdep` to map their names to debs.
+        The list is finally saved in the part's install directory.
+        """
+        cmd = []
+
+        # Clean up previously established list of packages in build snaps
+        cmd.append('rm -f "${CRAFT_PART_INSTALL}/.installed_packages.txt"')
+        cmd.append('rm -f "${CRAFT_PART_INSTALL}/.build_snaps.txt"')
+
+        if self._options.colcon_ros_build_snaps:  # type: ignore
```

Comment:
Indeed the file [`colcon_plugin.py`](https://github.com/artivis/snapcraft/blob/b76d8eb807835ed6a6bbc8dcd2482cb6704f8ce3/snapcraft/parts/plugins/colcon_plugin.py#L64) where the `ColconPluginProperties` is defined import this `_ros.py` file. It would thus results in a circular import.